### PR TITLE
Removal of custom vdump function and switch to logging module

### LINKED ISSUE #84: [funfuzz] Overhaul logging functionality
Currently, whatever primitive logging functionality is [triggered here](https://github.com/MozillaSecurity/funfuzz/blob/dadf874/util/subprocesses.py#L18).

We should move to a [logging module](https://github.com/MozillaSecurity/funfuzz/blob/bf54879/util/reposUpdate.py#L13), reposUpdate.py already supports this.

### PR DESCRIPTION
**Pull request by nth10sd**
Fairly mechanical switch away from our ancient custom `vdump` logging function in favour of the `logging` module.

This is a first take, I might add timestamps in the future, but first seeking feedback.

This fixes #84.

### CODE REVIEW

File: src/funfuzz/autobisectjs/autobisectjs.py [hunk 1]
```diff
@@ -38,6 +38,9 @@
     from pathlib import Path  # pylint: disable=import-error
     import subprocess
 
+AUTOBISECTJS_LOG = logging.getLogger("autobisectjs")
```
Comment (jschwartzentruber) [inline src/funfuzz/autobisectjs/autobisectjs.py]: For brevity you could just call this `LOG`. Up to you.

File: src/funfuzz/autobisectjs/autobisectjs.py [hunk 2]
```diff
@@ -127,9 +130,9 @@ def parseOpts():  # pylint: disable=invalid-name,missing-docstring,missing-retur
 
     # First check that the testcase is present.
     if "-e 42" not in options.parameters and not Path(options.runtime_params[-1]).expanduser().is_file():
-        print_(flush=True)
-        print_("List of parameters to be passed to the shell is: %s" % " ".join(options.runtime_params), flush=True)
-        print_(flush=True)
+        AUTOBISECTJS_LOG.info("\n")
```
Comment (jschwartzentruber) [inline src/funfuzz/autobisectjs/autobisectjs.py]: `"\n"` will put two lines in the log. You can pass an empty string `..._LOG.info("")` or log a separator like `"=" * 50` if you want this line to be more noticeable. This is a common pattern in this change, so I won't mark them all.

Comment (nth10sd) [reply src/funfuzz/autobisectjs/autobisectjs.py]: I like the `"=" * 50` separator idea, thanks!

Comment (nth10sd) [reply src/funfuzz/autobisectjs/autobisectjs.py]: (well at least for some instances)

Comment (nth10sd) [reply src/funfuzz/autobisectjs/autobisectjs.py]: I'll pass an empty string for least change now, but I'll keep in mind that separator idea for the future.

File: src/funfuzz/autobisectjs/autobisectjs.py (hunk 1 above)
Comment (nth10sd) [reply src/funfuzz/autobisectjs/autobisectjs.py]: I think I'd prefer to keep it as `AUTOBISECTJS_LOG`.

**Comment by nth10sd** [comment]
I'm working on removing support for Python 2.7 in a separate branch first. We might then have the option of tweaking the StreamHandler if necessary. (e.g. to edit the line ending)

**Comment by nth10sd** [comment]
This latest rebase on top of master, since it's 3.6+ only, has allowed us to tweak the StreamHandler to edit the terminator of a line.

**Pull request closed without merging**